scripts/BlackArts.lic

```diff
@@ -0,0 +1,7769 @@
+=begin
+    Alchemy script rewrite based on the original Tillman Alchemy script of DOOM!!!!!!1!!!11.
+    Currently relies on the following scripts for full utilization:
+      - bigshot, eherbs, eloot
+        author: elanthia-online
+  contributers: Deysh,
+          game: Gemstone
+          tags: alchemy
+       version: 1.0.0
+
+  Improvements:
+  v1.0.0 (2023-06-19)
+    - Initial creation
+=end
+
+require 'yaml'
+
+module BlackArts
+  BlackArts_version = '1.0.0'
+  @@data ||= nil
+
+  ##### Data & Setup Start #####
+
+  def self.data
+    @@data
+  end
+
+  def self.load(settings)
+    @@data = Data.new(settings)
+  end
+
+  def self.load_defaults()
+    default_hash = {}
+    Dir.mkdir("#{$data_dir}#{XMLData.game}") unless File.exist?("#{$data_dir}#{XMLData.game}")
+    Dir.mkdir("#{$data_dir}#{XMLData.game}/#{Char.name}") unless File.exist?("#{$data_dir}#{XMLData.game}/#{Char.name}")
+    Dir.mkdir("#{$data_dir}#{XMLData.game}/#{Char.name}/bigshot_profiles") unless File.exist?("#{$data_dir}#{XMLData.game}/#{Char.name}/bigshot_profiles")
+
+    File.write("#{$data_dir}#{XMLData.game}/#{Char.name}/blackarts.yaml", default_hash.to_yaml)
```

Comment:
Should use File.Join instead of string replacement.
```ruby
File.write(File.join(DATA_DIR, XMLData.game, Char.name, 'blackarts.yaml'), default_hash.to_yaml)
```